### PR TITLE
Add support for enabling Azure KeyVault KMS in aks-preview

### PR DESCRIPTION
**Pull request by bingosummer**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -3572,4 +3572,145 @@ def test_aks_create_with_network_plugin_none(self, resource_group, resource_grou
...
+        policy = self.cmd(set_policy, checks=[
+            self.check('properties.provisioningState', 'Succeeded')
+        ]).get_output_in_json()
+
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} ' \
+                     '--assign-identity {identity_id} ' \
+                     '--ssh-key-value={ssh_key_value} -o json'
+        self.cmd(create_cmd, checks=[
+            self.check('provisioningState', 'Succeeded')
+        ])
```
Comment (karataliu) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3700]: check enabled false and keyId empty here? to make below check more meaningful

File: src/aks-preview/azext_aks_preview/decorator.py [hunk 2]
```diff
@@ -1577,6 +1588,87 @@ def get_crg_id(self) -> str:
...
+            ):
+                enable_azure_keyvault_kms = self.mc.security_profile.azure_key_vault_kms.enabled
+
+        # this parameter does not need dynamic completion
+        # validation
+        if enable_validation:
+            if enable_azure_keyvault_kms and self._get_azure_keyvault_kms_key_id(enable_validation=False) is None:
+                raise RequiredArgumentMissingError(
+                    "--azure-keyvault-kms-key-id is not specified "
+                    "when --enable-azure-keyvault-kms is specified."
```
Comment (karataliu) [inline src/aks-preview/azext_aks_preview/decorator.py:1616]: align with message below?
--enable-azure-keyvault-kms requires --azure-keyvault-kms-key-id?

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (bingosummer) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3699]: add a check for security profile, which should be null.

File: src/aks-preview/azext_aks_preview/_help.py [hunk 3]
```diff
@@ -407,6 +407,12 @@
         - name: --message-of-the-day
           type: string
           short-summary: Path to a file containing the desired message of the day. Only valid for linux nodes. Will be written to /etc/motd.
+        - name: --enable-azure-keyvault-kms
+          type: bool
+          short-summary: (PREVIEW) Enable Azure KeyVault Key Management Service.
+        - name: --azure-keyvault-kms-key-id
```
Comment (bingosummer) [inline src/aks-preview/azext_aks_preview/_help.py:413]: The linter reports:
```
-  FAIL - HIGH severity: option_length_too_long
    Parameter: aks create, `azure_keyvault_kms_key_id` - The lengths of all options ['--azure-keyvault-kms-key-id'] are longer than threshold 22. Argument azure_keyvault_kms_key_id must have a short abbreviation.
    Parameter: aks create, `enable_azure_keyvault_kms` - The lengths of all options ['--enable-azure-keyvault-kms'] are longer than threshold 22. Argument enable_azure_keyvault_kms must have a short abbreviation.
    Parameter: aks update, `azure_keyvault_kms_key_id` - The lengths of all options ['--azure-keyvault-kms-key-id'] are longer than threshold 22. Argument azure_keyvault_kms_key_id must have a short abbreviation.
    Parameter: aks update, `enable_azure_keyvault_kms` - The lengths of all options ['--enable-azure-keyvault-kms'] are longer than threshold 22. Argument enable_azure_keyvault_kms must have a short abbreviation.
```

Do we add a short abbreviation or use `--enable-keyvault-kms` to make the length less than 22?

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/_help.py:413]: oh, you could suppress this warning in file `src/aks-preview/linter_exclusions.yml`.

File: src/aks-preview/azext_aks_preview/_help.py [hunk 4]
```diff
@@ -407,6 +407,12 @@
         - name: --message-of-the-day
           type: string
           short-summary: Path to a file containing the desired message of the day. Only valid for linux nodes. Will be written to /etc/motd.
+        - name: --enable-azure-keyvault-kms
+          type: bool
+          short-summary: (PREVIEW) Enable Azure KeyVault Key Management Service.
```
[the remark this answers is not in the document]
Comment (bingosummer) [reply src/aks-preview/azext_aks_preview/_help.py]: Thanks. Removed `(PREVIEW)`.

**Comment by bingosummer** [comment]
@karataliu @FumingZhang @zhoxing-ms could you please have another round of review?

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**